### PR TITLE
Bug 2000935: add volume mode selection in storage creation

### PR DESCRIPTION
**Pull request by shdn-ibm**
add a volume mode selection, to provide an option to use thin or thick volume
default setting is thick mode

image

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 1]
```diff
@@ -89,6 +103,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
           isRequired
         />
       </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Volume Mode')} fieldId="volmode-input">
+        <Select
+          onSelect={onModeSelect}
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:108]:
```suggestion
          onSelect={onModeSelect}
          id="volume-mode-input"
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 2]
```diff
@@ -89,6 +103,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
           isRequired
         />
       </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Volume Mode')} fieldId="volmode-input">
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:106]:
```suggestion
      <FormGroup label={t('ceph-storage-plugin~Volume Mode')} fieldId="volume-mode-input">
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 3]
```diff
@@ -31,6 +33,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
       : setEndpointValid(ValidatedOptions.error);
   };
 
+  const volModeList = ['thick', 'thin'];
+  const volModeOptions = volModeList.map((mode) => {
+    return <SelectOption key={mode} value={mode} />;
+  });
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:39]:
```suggestion
  const volModeOptions = VOLUME_MODES.map((mode) => <SelectOption key={mode} value={mode} />;
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 4]
```diff
@@ -31,6 +33,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
       : setEndpointValid(ValidatedOptions.error);
   };
 
+  const volModeList = ['thick', 'thin'];
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:36]:
```suggestion
  const  VOLUME_MODES = ['thick', 'thin'];
```
Define it outside the component

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 5]
```diff
@@ -31,6 +33,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
...
+  const volModeOptions = volModeList.map((mode) => {
+    return <SelectOption key={mode} value={mode} />;
+  });
+  const [isOpen, setIsOpen] = React.useState(false);
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:40]: Move it above with rest of the hooks.
Hoosk should always be at the top

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 6]
```diff
@@ -31,6 +33,18 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
...
+  const onToggle = () => setIsOpen(!isOpen);
+  const onModeSelect = (event, value) => {
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:42]: Add a space
```suggestion
  const onToggle = () => setIsOpen(!isOpen);
  
  const onModeSelect = (event, value) => {
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/type.ts [hunk 7]
```diff
@@ -37,4 +37,5 @@ export type FlashSystemState = {
   password: string;
   endpoint: string;
   poolname: string;
+  volmode?: string;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/type.ts:40]:
```suggestion
  volmode: string;
```
state should not have optional types

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 8]
```diff
@@ -31,6 +34,16 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
       : setEndpointValid(ValidatedOptions.error);
   };
 
+  const VOLUME_MODES = ['thick', 'thin'];
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:37]: define this outside the component

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 9]
```diff
@@ -89,6 +102,19 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
...
+        <Select
+          onSelect={onModeSelect}
+          id="volume-mode-input"
+          selections={formState.volmode}
+          onToggle={onToggle}
+          isOpen={isOpen}
+          isDisabled={false}
+          placeholderText={VOLUME_MODES[0]}
+        >
+          {volModeOptions}
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:115]:
```suggestion
          {VOLUME_MODES.map((mode) => <SelectOption key={mode} value={mode} />)}
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 10]
```diff
@@ -31,6 +34,16 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
       : setEndpointValid(ValidatedOptions.error);
   };
 
+  const VOLUME_MODES = ['thick', 'thin'];
+  const volModeOptions = VOLUME_MODES.map((mode) => <SelectOption key={mode} value={mode} />);
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:38]:
```suggestion
  const volModeOptions = VOLUME_MODES.map((mode) => <SelectOption key={mode} value={mode} />);
```
```suggestion

```

**Review by afreen23 — CHANGES_REQUESTED** [review]
Just some minors

**Review by afreen23 — APPROVED** [review]
/lgtm

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx [hunk 11]
```diff
@@ -89,6 +105,21 @@ export const FlashSystemConnectionDetails: React.FC<ExternalComponentProps<Flash
           isRequired
         />
       </FormGroup>
+      <FormGroup label={t('ceph-storage-plugin~Volume Mode')} fieldId="volume-mode-input">
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/ibm-flashsystem/index.tsx:108]:
```suggestion
      <FormGroup label={t('ceph-storage-plugin~Volume mode')} fieldId="volume-mode-input">
```

**Comment by SanjalKatiyar** [comment]
/retitle Bug 2000935: add volume mode selection in storage creation

**Pull request merged**